cocos/base/ccUtils.cpp

```diff
@@ -411,17 +411,27 @@ Node* findChild(Node* levelRoot, int tag)
 
 std::string getFileMD5Hash(const std::string &filename)
 {
+    Data data;
+    FileUtils::getInstance()->getContents(filename, &data);
+
+    return getDataMD5Hash(data);
+}
+
+std::string getDataMD5Hash(const Data &data)
+{
     static const unsigned int MD5_DIGEST_LENGTH = 16;
 
-    Data d;
-    FileUtils::getInstance()->getContents(filename, &d);
+    if (data.isNull() || data.getSize() == 0)
```

Comment:
`data.getSize()` isn't necessary here since `data.isNull()` will check the size.
```c++
bool Data::isNull() const
{
    return (_bytes == nullptr || _size == 0);
}
```